aiida_quantumespresso/workflows/pw/relax.py

```diff
@@ -256,8 +267,11 @@ def inspect_relax(self):
         if not prev_cell_volume:
             self.ctx.current_cell_volume = curr_cell_volume
 
-            # If meta convergence is switched off we are done
-            if not self.inputs.meta_convergence.value:
+            # If meta convergence is switched off, or the relax type doesn't change the volume we are done
+            if (
+                not self.inputs.meta_convergence.value or
+                RelaxType(self.inputs.relax_type) in [RelaxType.ATOMS, RelaxType.SHAPE]
+            ):
```

Comment:
Wonder if we should do the same here with `meta_convergence` as with `relax_type`. In `setup` we assign `self.ctx.meta_convergence` a boolean which is then used elsewhere in the logic. In principle it just assumes the value of the inputs but it is overridden based on the relax type. In this way we handle with this logic in one place and then can also have a report message in case the relax type overrides the meta_convergence input, so people are not super surprised when meta convergence is ignored